src/components/contract-boost-calculator.tsx

```diff
@@ -1,484 +1,632 @@
+		ihcMult,
+	]);
+
+	const boostDurationRaw = useMemo(() => {
+		const minDuration = boosts.reduce((min, b) => Math.min(min, b.durationMins), Infinity);
+		return minDuration * diliBonus;
+	}, [boosts, diliBonus]);
+
+	const boostDuration = useMemo(() => {
+		const mins = boostDurationRaw;
+
+		let boostDuration = mins >= 60 ? Math.floor(mins / 60) + 'hr' : '';
+		if (mins % 60 > 0) {
+			boostDuration += Math.round(mins % 60) + 'min';
+		}
+
+		return boostDuration;
+	}, [boostDurationRaw]);
+
+	const boostCost = useMemo(() => {
+		const sum = boosts.reduce((sum, b) => sum + b.geCost, 0);
+		return sum * 0.8;
+	}, [boosts]);
+
+	const totalLifeStones = useMemo(() => lifeCounts.reduce((sum, each) => sum + each), [lifeCounts]);
+	const totalDiliStones = useMemo(() => diliCounts.reduce((sum, each) => sum + each), [diliCounts]);
+
+	const enableOutput = totalLifeStones <= 12 && totalDiliStones <= 12;
+
+	// when any input in this section emits a `change` event, reset fetchState
+	// back to idle and then remove the event listener. because the useEffect
+	// has fetchState in the dependency list, when the button is clicked and
+	// fetchState becomes something other than IDLE, the event listener gets
+	// added (and then re-removed and re-added for every other state transition,
+	// but that's okay & i think unavoidable with this approach)
+	const listenerRef = useRef<HTMLElement>(null);
 	useEffect(() => {
-		// because equippedArtifactsByIGN is not typed, it will throw some crap
-		// eslint-disable-next-line @typescript-eslint/no-confusing-void-expression
-		const selectedIGN = equippedArtifactsByIGN.find(
-			(contributor: any) => contributor.isSelectedIGN,
-		);
-		if (selectedIGN) {
-			setFormState((prevState: any) => ({
-				...prevState,
-				IGN: (selectedIGN as { userName: string }).userName,
-			}));
+		const el = listenerRef.current;
+		if (!el) return;
+
+		const resetFetchState = () => {
+			calc.updateData({ fetchState: FetchState.IDLE });
+			el.removeEventListener('change', resetFetchState);
+		};
+
+		if (calc.data.fetchState !== FetchState.IDLE) {
+			el.addEventListener('change', resetFetchState);
 		}
-	}, [equippedArtifactsByIGN]);
 
+		return () => el.removeEventListener('change', resetFetchState);
+	}, [listenerRef, calc, calc.data.fetchState]);
+
+	// on reload, any active requests are canceled, so empty dependency array is on
+	// purpose: only set idle unconditionally on FIRST load
+	// eslint-disable-next-line react-hooks/exhaustive-deps
+	useEffect(() => calc.updateData({ fetchState: FetchState.IDLE }), []);
+
+	// TEMPORARY: override new vars for my beta testers. Remove before merge.
+	useEffect(
+		() =>
+			calc.updateData({
+				baseIhr: calc.data.baseIhr || '7440',
+				hatcheryCalm: calc.data.hatcheryCalm || '20',
+			}),
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+		[],
+	);
+
+	const resetExtras = useCallback(
+		() =>
+			calc.updateData({
+				doubleDuration: false,
+				baseIhr: '7440',
```

Comment:
btw why is this a string

Reply:
I should probably put a comment in `calculator` but these calc values are data storage for the controlled `input`s that edit them - so if they're stored as numbers, you can't backspace the last 0 from the input, which is super annoying (or, rephrased, you need to be able to represent both empty-input and input-contains-a-digit-0, and storing as `number` collapses that. I guess you could store as `number | undefined` but that doesn't actually gain you any ergonomics)